be/src/olap/row_cursor.h

```diff
@@ -161,6 +169,9 @@ class RowCursor {
     char* _variable_buf = nullptr;
     size_t _variable_len;
 
+    // for agg query, we don't need to finalize when scan agg object data
+    bool need_agg_finalize = true;
```

Comment:
Is it a good option to put this flag here? This struct only a row, it should not care how to assemble itself